git-changebar/src/gcb-plugin.c

```diff
@@ -49,6 +49,11 @@
 # define git_buf_dispose  git_buf_free
 # define git_error_last   giterr_last
 #endif
+#if ! defined (LIBGIT2_VER_MINOR) || ( (LIBGIT2_VER_MAJOR == 1) && (LIBGIT2_VER_MINOR < 4) )
+# define gcb_git_buf_reserved asize
```

Comment:
Since `asize` and `reserved` are rather generic words, I introduced a prefixed version to avoid accidental name clashes.

Is there a better way?

Reply:
Only putting `#if` around each use (there are only two AFAICT so it would be not too intrusive, but lots of trailing space removal noise might be hiding more) but this is fine here.

Reply:
I ruled out `if`ing the two occurrences pretty early. Since there are already compability macros in place, it seemed appropriate to follow this pattern here.

Regarding the whitespace changes: should I remove them from this PR?
For reviewing, you can also hide them in the "Files" tab here:
![github_files_no_whitespace](https://user-images.githubusercontent.com/617017/158699433-552a69e7-70f2-4fd4-9e67-32ffa778a60e.png)

Reply:
To be clear, my comment was not an objection, I was just noting the only __other__ way I can think of ... not a __better__ way.

Reply:
I appreciate the effort on not riddling the code with `#if`s, but the point of the compatibility code is to use new APIs and map it to old one where needed, so that the rest of the code is not using the deprecated/removed names.
Introducing a virtual name is not really doing the same, and kind of makes this awkward IMO, so here I'd rather have `#if`s carefully placed in the code, potentially with helper functions.